### DISCUSSION TITLE
Include tips on how to develop with docker

### DISCUSSION THREAD

**Discussion by WoosterInitiative** [cookiecutter/cookiecutter-django#4370] [Ideas]
One question that I haven't seen answered in the docs is how to do development using Docker, specifically how to handle migrations. There is a note about running `migrate` in the container, but no guidance on how to *create* the migrations (using `makemigrations`).

From what I've gathered, "best practice" would be to install Django on the dev machine and run `python -m manage makemigrations` and only *then* run the `migrate` command in the container.

I'm happy to create the PR for this, but I wanted to make sure that there is some agreement that this is the "right" way to do this?

**Comment by browniebroke** [comment]
> There is a note about running `migrate` in the container, but no guidance on how to _create_ the migrations (using `makemigrations`).

The [section](https://cookiecutter-django.readthedocs.io/en/latest/developing-locally-docker.html#execute-management-commands) you're referring to is titled "Execute Management Commands", and `makemigrations` _is_ a management command. There is nothing special about it, you can use the same as the other examples: `docker-compose -f local.yml run --rm django python manage.py makemigrations` 

In local development, the whole codebase is mounted as volume:

https://github.com/cookiecutter/cookiecutter-django/blob/fbc2373bcc8d7205541d37eb5e2934ace0fcc59d/%7B%7Bcookiecutter.project_slug%7D%7D/local.yml#L22-L23

Which means that migration files created in a container should appear on your host machine.
